File: 601-700/657-judge-circle.py
class Solution:
    def judgeCircle(self, moves: str) -> bool:
        # +1 to y for up(U) and vice versa
        y = 0
        # +1 to x for moving right and vice versa
        x = 0
        
        for move in moves:
            if move == "U":
                y += 1
            elif move == "D":
                y -= 1
            elif move == "R":
                x += 1
            elif move == "L":
                x -= 1
                
        return y == 0 and x == 0

File: 601-700/657-judge-circle.java

// O(n), 5ms, 39.4MB
class Solution {
  public boolean judgeCircle(String moves) {
      int vertical = 0;
      int horizontal = 0;
      
      for(char move : moves.toCharArray()) {
          if(move == 'U') {
              vertical++;
          } 
          
          if(move == 'D') {
              vertical--;
          }
          
          if(move == 'L') {
              horizontal--;
          }
          
          if(move == 'R') {
              horizontal++;
          }
      }
      
      return vertical == 0 && horizontal == 0;
  }
}